src/troubleshooting/content.md

```diff
@@ -0,0 +1,641 @@
+
+Let's start with Deployment and Service.
+
+## Connecting Deployment and Service
+
+The surprising news is that Service and Deployment aren't connected at all.
+
+Instead, the Service points to the Pods directly and skips the Deployment altogether.
+
+So what you should pay attention to is how the Pods and the Service are related to each other.
+
+You should remember three things:
+
+1. The Service selector should match the label of the Pod
+1. The Service `targetPort` should match the `containerPort` of the container inside the Pod
+1. The Service `port` can be any number. Multiple Service can use the same port because they have different IP addresses assigned.
+
+The following diagram summarises the how to connect the ports:
+
+```slideshow
+{
+  "description": "Connecting a Service and a Pod",
+  "slides": [
+    {
+      "image": "ports-1.svg",
+      "description": "Consider the following Pod exposed by a Service."
+    },
+    {
+      "image": "ports-2.svg",
+      "description": "When you create a Pod, you should define the port `containerPort` for each container in your Pods."
+    },
+    {
+      "image": "ports-3.svg",
+      "description": "When you create a Service, you can define a `port` and a `targetPort`. _But which one should you connect to the container?_"
+    },
+    {
+      "image": "ports-4.svg",
+      "description": "`targetPort` and `containerPort` should always match."
+    },
+    {
+      "image": "ports-5.svg",
+      "description": "If your container exposes port 3000, then the `targetPort` should match that number."
+    }
+  ]
+}
+```
+
+If you look at the YAML, the labels and `ports`/`targetPort` should match:
+
+```yaml|highlight=13-14,20,29,30-31|title=hello-world.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: app
+  labels:
+    track: canary
+spec:
+  selector:
+    matchLabels:
+      name: app
+  template:
+    metadata:
+      labels:
+        name: app
+    spec:
+      containers:
+      - name: app
+        image: learnk8s/app:1.0.0
+        ports:
+        - containerPort: 8080
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: app
+spec:
+  ports:
+  - port: 80
+    targetPort: 8080
+  selector:
+    name: app
+```
+
+_What about the `track: canary` label at the top of the Deployment?_
+
+_Should that match too?_
+
+That labels belong to the deployment, and it's not used by the Service's selector to route traffic.
+
+In other words, you can safely remove it or assign it a different value.
+
+_Assuming that you made the correct change, how do you test it?_
+
+You can check if the Pods have the right label with the following command:
+
+```terminal|command=1|title=bash
+kubectl get pods --show-labels
+```
+
+Or if you have Pods belonging to several applications:
+
+```terminal|command=1|title=bash
+kubectl get pods --selector name=app --show-labels
+```
+
+Where `name=app` is the label `name: app`.
+
+However, checking the label isn't enough.
```

Comment:
weird english: enough for what?

Maybe some enthusiastic wording to show that we can go deeper: "Checking the label is not all, we can do a real connection!"

Also, port-forward is awesome, why not do it for pods too, so we can determine where is the problem: in the container/pod/deployment configuration or in the service configuration.